csvCreator.js

```diff
@@ -0,0 +1,126 @@
+const fs = require('fs');
+const csvWriter = require('csv-write-stream');
+var writer = csvWriter();
+var faker = require('faker');
+require('events').EventEmitter.defaultMaxListeners = 100000;
+
+// const listingsGen = () => {
+//   let i = 0
+
+//   const NUM_LISTINGS = 1000;
+
+//   writer.pipe(fs.createWriteStream('listings-data.csv'));
+//   function write() {
+//     var ok = true;
+//     do{
+//       i++;
+//       ok = writer.write({
+//         listingId: i,
+//         listingName: faker.lorem.word(),
+//         listingDescription: faker.lorem.words(),
+//         listingLocation: faker.address.city(),
+//         listingStars: Math.ceil(Math.random() * 5),
+//         listingNumReviews: Math.floor(Math.random() * 100)
+//       });
+//     } while(i < NUM_LISTINGS && ok);
+//     if(i < NUM_LISTINGS) {
+//       console.log('in here')
+//       writer.once('drain', write);
+//     }
+//   }
+//   write();
+//   writer.end();
+//   console.log('done', i);
+// };
+const listingsGen = () => {
+  writer.pipe(fs.createWriteStream('users-data.csv'));
+  for (var i = 0; i < 1000000; i++) {
```

Comment:
I would say maybe start with a little bit lower number just for initial tests. If you already tested with lower numbers then ignore this.